aws-http-nio-client-netty/pom.xml

```diff
@@ -50,6 +50,8 @@
         <dependency>
             <groupId>io.netty</groupId>
             <artifactId>netty-transport-native-epoll</artifactId>
+            <version>${netty.version}</version>
+            <classifier>linux-x86_64</classifier>
```

Comment:
You're right, it's not possible without reflection but given we can do it once for the default case I don't think that's a big hit. 

This is the classic definition of a runtime dependency - I don't think we should get in the business of supplying every native implementation for each architectural variant. We've pushed back on this in the past.

Nio as the default is still fairly efficient, we can add documentation to explain how to pull in Epoll / KQueue / some windoze implementation. Point is it's so dependent on how the customer is going to deploy their application that I don't think we should make it a compile-time dependency.